app/shared/toolbar/toolbar.component.ts

```diff
@@ -69,6 +70,30 @@ export class ToolbarComponent implements OnInit {
     this.authService.addListener(this);
     this.currentUser = User.getEmptyUser();
     this.onChange();
+
+    // Translate: set default language
+    this.translate.use('de');
+  }
+
+  // Translate: check if the selected lang is current lang
+  isCurrentLang(lang: string) {
+      return lang === this.translate.currentLang;
+  }
+
+  // Translate: Language Toggler with respect to selected language
+  selectLang() {
+      for (let lang of this.supportedLangs) {
+          lang.active = !lang.active;
+          if (lang.active) {
+              this.translate.use(lang.value);
+          }
+      }
+      this.refreshText();
+  }
+
+  // Translate: Refresh translation when language change. This is used if Translate service is used instead of Pipe
+  refreshText() {
```

Comment:
This could be used, if we want translations as dynamic. The commented out code shows how to use it. So, if the requirement comes we can easily set up.